util/reflect.go

```diff
@@ -124,6 +123,21 @@ func IsValueScalar(v reflect.Value) bool {
 	return !IsValueStruct(v) && !IsValueMap(v) && !IsValueSlice(v)
 }
 
+// IsInterfaceToStructPtr reports whether v is an interface that contains a pointer
+// to a struct.
+func IsValueInterfaceToStructPtr(v reflect.Value) bool {
+	return IsValueInterface(v) && IsValueStructPtr(v.Elem())
+}
+
+// PtrToValue returns the dereferenced reflect.Value of value if it is a ptr, or
+// value if it is not.
+func PtrToValue(value reflect.Value) reflect.Value {
```

Comment:
This is currently not covered in test coverage per https://coveralls.io/builds/13569303/source?filename=util%2Freflect.go

Reply:
I was meaning to take this out, it's not general enough for this package.